src/pkg/task/hook.go

```diff
@@ -37,31 +45,63 @@ type HookHandler struct {
 }
 
 // Handle the job status changing webhook
-func (h *HookHandler) Handle(ctx context.Context, taskID int64, sc *job.StatusChange) error {
-	task, err := h.taskDAO.Get(ctx, taskID)
+func (h *HookHandler) Handle(ctx context.Context, sc *job.StatusChange) error {
+	logger := log.GetLogger(ctx)
+
+	jobID := sc.JobID
+	if sc.Metadata.JobKind == job.KindScheduled {
+		jobID = sc.Metadata.UpstreamJobID
```

Comment:
For the job with `KindScheduled `, if it is not scheduled by the periodical job, `KindScheduled ` may be empty.

Reply:
Yes, for the jobs whose kind isn't `KindScheduled`, read the job ID from `StatusChange.JobID`

Reply:
What I mean, the following code may make the jobID is empty under the case I mentioned

```
if sc.Metadata.JobKind == job.KindScheduled {
		jobID = sc.Metadata.UpstreamJobID
```

Reply:
```
if sc.Metadata.JobKind == job.KindScheduled {
  // Scheduled by a periodical job
   if len(sc.Metadata.UpstreamJobID) > 0{
       jobID = sc.Metadata.UpstreamJobID
   }
}
```

Reply:
@ywk253100

Reply:
updated. Thanks